.github/workflows/ci.yml

```diff
@@ -196,3 +196,86 @@ jobs:
           endtoend_suite: versioned-admin
           endtoend_config: vendor/silverstripe/versioned-admin/behat.yml
           endtoend_tags: job2,job3,job4,job5
+  checklicenses:
+    name: Check licenses
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: read
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+      - name: Get PHP version
+        id: phpversion
+        run: |
+          # Get the PHP version to use from composer.json
+          PHP=$(jq -r '.require["php"]' composer.json)
+          # Remove the leading caret
+          PHP=${PHP//^/}
+          echo "::set-output name=version::$PHP"
+      - name: Install PHP
+        uses: shivammathur/setup-php@c541c155eee45413f5b09a52248675b1a2575231 # v2.31.1
+        with:
+          php-version: ${{ steps.phpversion.outputs.version }}
+          extensions: curl, dom, gd, intl, json, ldap, mbstring, mysql, tidy, xdebug, zip
+          tools: composer:v2
+          coverage: xdebug
+      - name: Composer install
+        run: composer install
+      - name: Check licenses
+        run: |
+          # Validate licenses of all composer dependencies are allowed
+          echo "Checking licenses of all dependencies"
+          composer global require madewithlove/license-checker
+          COMPOSER_GLOBAL_HOME=$(composer -q -n config --global home)
+          # Fetch a list of allowed SPDX identifiers from the repository
+          URL=https://raw.githubusercontent.com/silverstripe/gha-run-tests/refs/heads/1/allowed-spdx-delimited.txt
+          echo "Fetching from $URL"
+          SPDX_ALLOWED_DELIMITED=$(curl -s $URL)
+          echo "SPDX_ALLOWED_DELIMITED is $SPDX_ALLOWED_DELIMITED"
+          if [[ $SPDX_ALLOWED_DELIMITED =~ "404: Not Found" ]]; then
```

Comment:
This should be removed before being merged, I've got it here simply because https://github.com/silverstripe/gha-run-tests/pull/44 hasn't been merged